x-pack/plugins/cloud/server/plugin.ts

```diff
@@ -71,6 +71,21 @@ export interface CloudSetup {
     url?: string;
     secretToken?: string;
   };
+  /**
+   * `true` when running on Serverless Elastic Cloud
+   * Note that `isCloudEnabled` will always be true when `isServerlessEnabled` is.
+   */
+  isServerlessEnabled: boolean;
+  /**
+   * Serverless configuration
+   */
+  serverless: {
+    /**
+     * The serverless projectId.
+     * Will always be present if `isServerlessEnabled` is `true`
+     */
+    projectId?: string;
```

Comment:
NIT I wonder if it should be the whole `'serverless'` section that is optional instead.
Does it make sense to have a `serverless: {}` empty section?

Reply:
For now it would make more sense. The thing is, we will likely add more serverless attributes to this section, and given their presence may be optional, it could complicate the thing (the same reason why the props are each individually nullable for ESS-related props)